dev/com.ibm.ws.jsp.2.3_fat/fat/src/com/ibm/ws/jsp23/fat/tests/JSPTests.java

```diff
@@ -802,6 +836,34 @@ public void testEDR() throws Exception {
         LOG.info("Servlet response : " + response2.getText());
         assertTrue("The response did not contain: " + expect2, response2.getText().contains(expect2));
         server.deleteFileFromLibertyServerRoot(relEdrPath + orgEdrFile); // cleanup
+        Thread.sleep(500L); // ensure file is deleted
+    }
+
+    public void makeConcurrentRequests(WebConversation wc1, WebRequest request1, int numberOfCalls) throws Exception {
+        final ExecutorService executor = Executors.newFixedThreadPool(numberOfCalls);
+        final Collection<Future<Boolean>> tasks = new ArrayList<Future<Boolean>>();
+
+        // run the test multiple times concurrently
+        for (int i = 0; i < numberOfCalls; i++) {
+            tasks.add(executor.submit(new Callable<Boolean>() {
+                @Override
+                public Boolean call() throws Exception {
+                    LOG.info("Thread Started: Making Request!");
+                    wc1.getResponse(request1);
+                    return true;
+                }
+            }));
+        }
+        
+        // check runs completed successfully
+        for (Future<Boolean> task : tasks) {
+            try {
+                if (!task.get())
+                    throw new Exception("0");
```

Comment:
are the `"0"` and `"1"` just used for differentiation?

Reply:
I think so. I left this alone from the other example, but I can change it if you prefer? 

https://docs.oracle.com/javase/9/docs/api/java/util/concurrent/Future.html#get--